src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -221,6 +225,46 @@ public GetTransactionModel GetTransaction(string txid)
             return model;
         }
 
+        [ActionName("listunspent")]
+        [ActionDescription("Returns an array of unspent transaction outputs belonging to this wallet.")]
+        public UnspentCoinModel[] ListUnspent(int minConfirmations = 1, int maxConfirmations = 9999999, string addressesJson = null)
+        {
+            List<BitcoinAddress> addresses = new List<BitcoinAddress>();
+            if (!string.IsNullOrEmpty(addressesJson))
+            {
+                JsonConvert.DeserializeObject<List<string>>(addressesJson).ForEach(i => addresses.Add(BitcoinAddress.Create(i, this.fullNode.Network)));
+            }
+            var accountReference = this.GetAccount();
+            IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager.GetSpendableTransactionsInAccount(accountReference, minConfirmations);
+
+            var unspentCoins = new List<UnspentCoinModel>();
+            foreach (var spendableTx in spendableTransactions)
+            {
+                int confirmations = spendableTx.Transaction.BlockHeight == null ? 0 : this.chain.Tip.Height - spendableTx.Transaction.BlockHeight.Value;
+                if (confirmations <= maxConfirmations)
+                {
+                    if (!addresses.Any() || addresses.Contains(BitcoinAddress.Create(spendableTx.Address.Address, this.fullNode.Network)))
+                    {
+                        unspentCoins.Add(new UnspentCoinModel()
+                        {
+                            Account = accountReference.AccountName,
+                            Address = spendableTx.Address.Address,
+                            Id = spendableTx.Transaction.Id,
+                            Index = spendableTx.Transaction.Index,
+                            Amount = spendableTx.Transaction.Amount,
+                            ScriptPubKeyHex = spendableTx.Transaction.ScriptPubKey.ToHex(),
+                            RedeemScriptHex = spendableTx.Transaction.ScriptPubKey.PaymentScript?.ToHex(), // Is this right?
+                            Confirmations = confirmations,
+                            IsSpendable = spendableTx.Transaction.IsSpendable(),
+                            IsSolvable = spendableTx.Transaction.IsSpendable() // Is it's spendable we assume it's solvable.
```

Comment:
Seems like a reasonable assumption - if the transaction was sent to an address we own we at least nominally know (knew) how to construct the redeem scripts etc., presuming wallet support for such transactions.

There are probably edge cases, but those can be addressed in future.

Reply:
Yes this IsSolvable is a fairly new feature to bitcoin core so not sure exactly the purpose behind it.